packages/mattermost-redux/src/reducers/entities/posts.ts

```diff
@@ -285,6 +286,21 @@ function handlePostReceived(nextState: any, post: Post) {
         Reflect.deleteProperty(nextState, post.pending_post_id);
     }
 
+    const rootPost: Post = nextState[post.root_id];
+    if (post.root_id && rootPost) {
+        const participants = rootPost.participants || [];
+        const nextRootPost = {...rootPost};
+        if (!participants.find((user: UserProfile) => user.id === post.user_id)) {
+            nextRootPost.participants = [...participants, {id: post.user_id}];
+        }
+
```

Comment:
Would it be possible to handle removal similar to how we do it here in a reducer?

From the deleted post's rootId we could find the threadId and then update the participants if the deleted post was the only post by the user.

Reply:
Unfortunately @ashishbhate, we can't get the thread state in the posts reducer so the only way would be to loop through every single post to check if it's root_id is the same. I don't think that would be too bad since deletes shouldn't happen all that often.

Either way I think that's a for a different ticket, not this one.

Reply:
> Either way I think that's a for a different ticket, not this one.

👍